notebooks/roc_plot_generator.py

```diff
@@ -548,25 +633,7 @@ def get_points(corrs):
 
 if "16H" in methods:
     if "16H" not in points: points["16H"] = []
-    points["16H"].extend(get_points(sixteen_heads_corrs))
-
-#%%
-
-def discard_non_pareto_optimal(points):
-    ret = [(0.0, 0.0), (1.0, 1.0)]
-    for x, y in points:
-        for x1, y1 in points:
-            if x1 <= x and y1 >= y and (x1, y1) != (x, y):
-                break
-        else:
-            ret.append((x, y))
-    return ret
-
-for method in methods:
-    processed_points = discard_non_pareto_optimal(points[method]) #  for key in points[method]}
-    # sort by x
-    processed_points = sorted(processed_points, key=lambda x: x[0])  # for k in processed_points}
-    points[method] = processed_points
```

Comment:
I think that removing i) non-Pareto optimal points and ii) not sorting is probably causing the issue

Reply:
Yes, that's right. I don't think we should do that naively though, because in the original context of ROC curves, it's just by varying the threshold that the line is drawn.

An acceptable compromise is to do the Pareto thing but also make a scatter plot of each run.

This `discard_non_pareto_optimal` if anything should be moved to the plotly-creation file

Reply:
Cool, I agree that moving sounds reasonable and so does making scatter plots (probably for the Appendix). 

It should be fairly easy to create scatter plots:

You'll want to edit t[his line](https://github.com/ArthurConmy/Automatic-Circuit-Discovery/blob/274cbe9/notebooks/plotly_roc_plot.py#L142) to be markers in order to get a scatter plot:

![Screenshot 2023-05-17 at 10 27 39](https://github.com/ArthurConmy/Automatic-Circuit-Discovery/assets/35957051/ec1584f5-80e8-4560-8b11-2a74fbbdae14)

Reply:
I'm happy to do this if you want?

Reply:
I did this following your instructions, thank you :)